services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/service/session/SwitchSessions.java

```diff
@@ -0,0 +1,67 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.service.session;
+
+import net.floodlightcontroller.core.IOFSwitch;
+import org.projectfloodlight.openflow.protocol.OFMessage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+class SwitchSessions {
+    private final Map<Long, Session> sessionsByXid = new HashMap<>();
```

Comment:
Better leave it with HashMap. FL design let us to leave some methods without synchronization (always called from single/master thread). Plus in case of `ConcurrentHashMap` will make double lock/synchronization in method `unbindSession`.